src/cloudai/systems/slurm/strategy/slurm_command_gen_strategy.py

```diff
@@ -57,22 +57,41 @@ def __init__(self, system: SlurmSystem, cmd_args: Dict[str, Any]) -> None:
         )
         self.docker_image_url = self.cmd_args.get("docker_image_url", "")
 
-    def _format_env_vars(self, env_vars: Dict[str, Any]) -> str:
-        """
-        Format environment variables for inclusion in a batch script.
+    def gen_exec_command(self, tr: TestRun) -> str:
+        env_vars = self._override_env_vars(self.system.global_env_vars, tr.test.extra_env_vars)
+        cmd_args = self._override_cmd_args(self.default_cmd_args, tr.test.cmd_args)
+        slurm_args = self._parse_slurm_args(
+            tr.test.test_template.__class__.__name__, env_vars, cmd_args, tr.num_nodes, tr.nodes
+        )
 
-        Args:
-            env_vars (Dict[str, Any]): Environment variables to format.
+        if tr.prologue:
```

Comment:
It is a bit hard to read because of nested and repeated conditions. Let's try to simplify to something like this:
```py
        srun_command = self._gen_srun_command(slurm_args, env_vars, cmd_args, tr.test.extra_cmd_args)
        command_list: list[str] = []
        indent = ""
        if tr.prologue:
            command_list = [self.gen_prologue(tr.prologue, tr.output_path), "if [ $PROLOGUE_SUCCESS -eq 1 ]; then"]
            indent = "    "

        command_list.append(f"{indent}{srun_command}")

        if tr.epilogue:
            command_list.append(f"{indent}{self.gen_epilogue(tr.epilogue, tr.output_path)}")

        if tr.prologue:
            command_list.append("fi")
```